### DISCUSSION TITLE
ImportError: no module named 'adafruit_mlx90614'

### DISCUSSION THREAD

**Discussion by chadoulis** [STELLA-Landsat/STELLA#27] [Build Notes: How do I make a STELLA?]
I bought the `MLX90614ESF-BCC-000-SP-ND` sensor instead of the `MLX90614ESF-BCC-000-TU`. When loading the code.py file I am getting the following error

```
Traceback (most recent call last):
  File "code.py", line 47, in <module>
ImportError: no module named 'adafruit_mlx90614'
```

Tried installing the `adafruit-circuitpython-mlx90614` library, following [this tutorial](https://codewith.mu/en/tutorials/1.2/pypi). I still get the same error. Any help would be highly appreciated.

**Comment by paulmirel** [comment]
The SP-ND suffix is just packaging and DigiKey nomenclature. It's the same exact sensor. 

ImportError only happens when a library file is not in the lib folder on the device. 

Oh. I see. Somewhere, the filename got changed. Notice the error is looking for adafruit_mlx90614, and the file you've installed is called adafruit-circuitpython-mlx90614 That discrepancy will prevent it loading. 

Looking into it, that tutorial you followed does not lead you to correct results. 

Please use the contents of the lib folder from the STELLA download. That has the library in it that the software is expecting. 

I show that library import on line 29, not line 47. What software version are you running? Most up to date version is v2.2.0

Here's the latest code.py and lib: 
[S2_C_L.zip](https://github.com/STELLA-Landsat/STELLA/files/11729499/S2_C_L.zip)

Your STELLA may not run properly because the code is expecting the on-board sensors, which you do not have. That's a software mod I hadn't planned, but I'll see if I can find some time. Best case scenario is that the program returns null for those values but keeps running anyway.

**Comment by paulmirel** [comment]
OHHHH. That tutorial you linked to is for downloading and installing libraries that run on the host computer, not on the microcontroller. (You can run python in the Mu editor on the host computer) You need the .mpy style libraries for use with the microcontroller.

**Reply by chadoulis** [reply]
The issue was that I had forgotten to copy the `lib` folder and paste it into the `CIRCUITPY` drive after one of the many hard resets I did. Thank you.

**Comment by paulmirel** [comment]
Excellent. Thanks for letting us know!
